src/components/tooltip/daikin-tooltip.ts

```diff
@@ -0,0 +1,187 @@
+import {
+  autoUpdate,
+  computePosition,
+  flip,
+  offset,
+  shift,
+} from "@floating-ui/dom";
+import { cva } from "class-variance-authority";
+import { css, html, LitElement, unsafeCSS, type PropertyValues } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import { createRef, ref, type Ref } from "lit/directives/ref.js";
+import tailwindStyles from "../../tailwind.css?inline";
+
+const cvaTooltip = cva(
+  [
+    "absolute",
+
+    "p-3",
+    "justify-center",
+    "items-center",
+    "rounded",
+    "w-max",
+    "top-0",
+    "left-0",
+    "max-w-[312px]",
+
+    "text-sm",
+    "font-daikinSerif",
+    "font-normal",
+    "not-italic",
+    "leading-5",
+  ],
+  {
+    variants: {
+      variant: {
+        light: [
+          "border",
+          "border-solid",
+          "border-daikinNeutral-800",
+          "bg-[#ffffffe6]",
+          "text-black",
+        ],
+        dark: ["bg-[#414141e6]", "text-white"],
+      },
+      open: {
+        true: ["visible", "opacity-100"],
+        false: ["invisible", "opacity-0"],
+      },
+    },
+  }
+);
+
+/**
+ * A tooltip component.
+ *
+ * @slot - A slot for the element to which the tooltip is attached (the trigger element).
+ * @slot tooltip - A slot for the tooltip content.
+ */
+@customElement("daikin-tooltip")
+export class DaikinTooltip extends LitElement {
+  static override readonly styles = css`
+    ${unsafeCSS(tailwindStyles)}
+
+    :host {
+      display: inline-block;
+    }
+  `;
+
+  /**
+   * Specifies the position of the tooltip relative to the trigger.
+   */
+  @property({ reflect: true, type: String })
+  placement: "top" | "bottom" | "left" | "right" = "bottom";
+  /**
+   * Specifies tooltip type
+   */
+  @property({ reflect: true, type: String })
+  variant: "light" | "dark" = "light";
+  /**
+   * Specifies whether the tooltip should be open when it first renders
+   */
+  @property({ reflect: true, type: Boolean })
+  open = false;
+  /**
+   * Specifies the content of the tooltip. Ignored if the tooltip slot exists.
+   */
+  @property({ reflect: true, type: String })
+  description = "";
+  /**
+   * if true, the tooltip will hide on click
+   */
+  @property({ reflect: true, type: Boolean })
+  closeOnClick = false;
+
+  private _tooltipRef: Ref<HTMLElement> = createRef();
+
+  private _triggerRef: Ref<HTMLElement> = createRef();
+
+  private _cleanUpAutoUpdate: (() => void) | null = null;
+
+  private _startAutoUpdate() {
+    const reference = this._triggerRef.value;
+    const float = this._tooltipRef.value;
+    if (!reference || !float) {
+      return;
+    }
+    this._cleanUpAutoUpdate?.();
+    this._cleanUpAutoUpdate = autoUpdate(reference, float, () => {
+      computePosition(reference, float, {
+        placement: this.placement,
+        middleware: [offset({ mainAxis: 20 }), flip(), shift()],
```

Comment:
It might be possible to make it customizable.

```suggestion
        middleware: [offset({ mainAxis: spacing }), flip(), shift()],
```

where

```ts
const hostStyles = window.getComputedStyles(this);
const spacing = parseInt(hostStyles.getPropertyValue("--dds-tooltip-spacing") || DEFAULT_TOOLTIP_SPACING, 10);
```